storage/storage.go

```diff
@@ -69,40 +71,52 @@ func NewUniqueObjectIterator(iter1, iter2 ObjectIterator) ObjectIterator {
 var _ ObjectIterator = (*uniqueObjectIterator)(nil)
 
 // Next returns the next most unique object from the two underlying iterators.
-func (u *uniqueObjectIterator) Next() (*openfgapb.Object, error) {
-
+// If the context is cancelled or times out, it should return ErrIteratorDone
+func (u *uniqueObjectIterator) Next(ctx context.Context) (*openfgapb.Object, error) {
+ForLoop:
 	for {
-		obj, err := u.iter1.Next()
-		if err != nil {
-			if err == ErrIteratorDone {
-				break
+		select {
+		case <-ctx.Done():
```

Comment:
Why are these necessary? The underlying iterators respect the context cancellation and this simple wrapper ontop of them will inherit the cancellation automatically.

Reply:
For example, `u.iter1.Next(ctx)` and `u.iter2.Next(ctx)` should both be cancelled when the parent `ctx` is cancelled, and so the `ctx.Done()` herein should be redundant

Reply:
In general our iterators should probably be adapted to be more async to handle context cancellation.

https://pauldigian.com/advanced-go-make-synchronous-code-asynchronous-with-contextcontext-and-channels